### PR TITLE
DEPR: pandas.core.index

### LINKED ISSUE #19711: DEPR: remove pandas.core.index
this is an old artifice which needs to be deprecated. It might be used somewhat internally. These usages need to be moved to pandas.core.indexes.api

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #19711
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v1.0.0.rst [hunk 1]
```diff
@@ -490,7 +490,7 @@ Deprecations
 - :func:`eval` keyword argument "truediv" is deprecated and will be removed in a future version (:issue:`29812`)
 - :meth:`Categorical.take_nd` is deprecated, use :meth:`Categorical.take` instead (:issue:`27745`)
 - The parameter ``numeric_only`` of :meth:`Categorical.min` and :meth:`Categorical.max` is deprecated and replaced with ``skipna`` (:issue:`25303`)
--
+- :module:`pandas.core.index` has been deprecated and will be removed in a future version, use :module:`pandas.core.indexes.api` instead (:issue:`19711`)
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v1.0.0.rst]: This should not be the recommended alternative, I think? (nothing in `pandas.core` should be public)

I think @jreback's suggestion to use `pandas.core.indexes` was for internal replacement

File: pandas/core/index.py [hunk 2]
```diff
@@ -19,3 +21,9 @@
     get_objs_combined_axis,
 )
 from pandas.core.indexes.multi import _sparsify  # noqa:F401
+
+warnings.warn(
+    "pandas.core.index is deprecated and will be removed in a future version.  "
+    "Use pandas.core.indexes.api instead.",
+    FutureWarning,
```
Comment (jorisvandenbossche) [inline pandas/core/index.py]: Let's start with a DeprecationWarning?

Comment (jreback) [reply pandas/core/index.py]: this  has been around forever 
let’s just be loud about it (and it had been deprecation via documentation. forever)

File: doc/source/whatsnew/v1.0.0.rst (hunk 1 above)
Comment (jbrockmendel) [reply doc/source/whatsnew/v1.0.0.rst]: so remove all of this note or just the part about using indexes.api?

Comment (jbrockmendel) [reply doc/source/whatsnew/v1.0.0.rst]: the docbuild doesnt like ":module:" here, what is the appropriate label?

Comment (jorisvandenbossche) [reply doc/source/whatsnew/v1.0.0.rst]: Nothing, just use double backticks for code. The `:obj:` things are only to link to something (like an API page), but for a module we have nothing to link to.

Comment (jorisvandenbossche) [reply doc/source/whatsnew/v1.0.0.rst]: > so remove all of this note or just the part about using indexes.api?

You can mention that all public classes (eg the several index classes) are available from the top-level namespace.

Comment (jbrockmendel) [reply doc/source/whatsnew/v1.0.0.rst]: thanks

**Review by WillAyd — APPROVED** [review]
lgtm. Can you fix up merge conflict?

File: pandas/core/index.py [hunk 3]
```diff
@@ -19,3 +21,9 @@
     get_objs_combined_axis,
 )
 from pandas.core.indexes.multi import _sparsify  # noqa:F401
+
+warnings.warn(
+    "pandas.core.index is deprecated and will be removed in a future version.  "
+    "Use pandas.core.indexes.api instead.",
```
Comment (jorisvandenbossche) [inline pandas/core/index.py]: Can you update this with the same as what you put in the whatsnew? (that is is available from the top-level namespace)

**Review by jorisvandenbossche — CHANGES_REQUESTED** [review]

File: doc/source/whatsnew/v1.0.0.rst [hunk 4]
```diff
@@ -453,7 +453,7 @@ See :ref:`install.dependencies` and :ref:`install.optional_dependencies` for mor
 Other API changes
 ^^^^^^^^^^^^^^^^^
 
-- Bumpded the minimum supported version of ``s3fs`` from 0.0.8 to 0.3.0 (:issue:`28616`)
+- Bumped the minimum supported version of ``s3fs`` from 0.0.8 to 0.3.0 (:issue:`28616`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.0.0.rst]: can you update the section that has our min supported version (in the whatsnew) (don't really need this issue line in that case)

**Review by jorisvandenbossche — APPROVED** [review]

File: pandas/core/index.py (hunk 2 above)
Comment (jorisvandenbossche) [reply pandas/core/index.py]: OK, opened an issue about it at fastparquet: https://github.com/dask/fastparquet/issues/470

**Pull request merged**